test/e2e/framework/pods.go

```diff
@@ -70,3 +74,23 @@ func (f *Framework) MungePodSpec(pod *api.Pod) {
 		pod.Spec.NodeName = TestContext.NodeName
 	}
 }
+
```

Comment:
Add comments describing what this function does. The arg `update` may need some more explanation.

nit: rename `update` to `updateFn` for clarity?